qemu/tests/qemu_guest_agent.py

```diff
@@ -1554,6 +1555,125 @@ def setup(self, test, params, env):
+                if self.fswrite_thread.isAlive():
+                    session2 = self._get_session(self.params, None)
+                    self._open_session_list.append(session2)
+                    k_cmd = "wmic process where \"name='python.exe' and " \
+                            "CommandLine Like '%fsfreeze%'\" call terminate"
+                    s, o = session2.cmd_status_output(k_cmd)
+                    if s:
+                        self.test.error("Command '%s' failed, status: %s, "
+                                        "output: %s" % (k_cmd, s, o))
+            error_context.context("Check guest FS status.", logging.info)
+            # init fs status to 'thaw'
+            fs_status = "thaw"
+            file_name = "/tmp/fsfreeze_%s.txt" % flag
+            process.system("rm -rf %s" % file_name)
+            self.vm.copy_files_from("C:\\fsfreeze.txt", file_name)
+            with open(file_name, 'r') as f:
+                list_time = f.readlines()
+
+            for i in list(range(0, len(list_time))):
+                list_time[i] = list_time[i].strip()
+
+            for i in list(range(1, len(list_time))):
+                num_d = float(list_time[i]) - float(list_time[i - 1])
+                if num_d > 8:
+                    logging.info("Time stamp is not continuous,"
+                                 " so the FS is frozen.")
+                    fs_status = "frozen"
+                    break
+            if not fs_status == flag:
+                self.test.fail("FS is not %s, it's %s." % (flag, fs_status))
+
+        error_context.context("Check guest agent command "
+                              "'guest-fsfreeze-freeze/thaw'",
+                              logging.info)
+        # make write time longer than freeze timeout
+        write_timeout = int(params["freeze_timeout"]) + 10
+        try:
+            expect_status = self.gagent.FSFREEZE_STATUS_THAWED
+            self.gagent.verify_fsfreeze_status(expect_status)
+        except guest_agent.VAgentFreezeStatusError:
+            # Thaw guest FS if the fs status is incorrect.
+            self.gagent.fsthaw(check_status=False)
+
+        background_start()
+        error_context.context("Freeze the FS.", logging.info)
+        self.gagent.fsfreeze()
+        try:
+            result_check("frozen", write_timeout)
+            # Next, thaw guest fs.
+            background_start()
+            error_context.context("Thaw the FS.", logging.info)
+            try:
+                self.gagent.fsthaw()
+            except guest_agent.VAgentCmdError as detail:
+                if re.search("fsfreeze is limited up to 10 seconds", str(detail)):
+                    logging.info("FS is thaw as it's limited up to 10 seconds.")
+                else:
+                    test.fail("guest-fsfreeze-thaw cmd failed with:"
+                              "('%s')" % str(detail))
+        except Exception:
+            # Thaw fs finally, avoid problem in following cases.
```

Comment:
Well, the comment seems mismatch the code, if you want thaw fs anyway, should execute it within `finally:`. If I understand correctly, you try to start write process when fsfreeze at line 1656, then here we actually expect it fail->raise Exception, then we `thaw` it here. Instead of at line 1659.
```
self.gagent.fsfreeze()
try:
    # write action
except Exception:
    # here is what we want, write should not sucess
    # So thaw here
    try:
         self.gagent.fsthaw()
    except ...:
         if re.search():
             logging.info()
         else:
             test.fail()
else:
    # that's not what we want, the second write success!
    test.fail()
finally:
    # thaw again if you want.
```

Reply:
It's not like you said. 
The test steps are like this.
1. background_start()
2. freeze cmd
3. stop background and result_check() 
4. background_start()
5. thaw cmd
6. stop background and result_check()

Reply:
@xiagao Thanks for explanation, I didn't notice that the guest will automatically thaw after 10 seconds before first thaw() is actually sent, clear now.